uber/site_sections/registration.py

```diff
@@ -801,6 +801,54 @@ def review(self, session):
                                     .order_by(Attendee.full_name).all()}
 
     @site_mappable
+    def promo(self, session, message='', **params):
+        pc = session.promo_code(params)
+        now = datetime.now(UTC)
+        today = "%d-%02d-%02d" % (now.date().year, now.date().month, now.date().day)
+        max_day = "%d-%02d-%02d" % (c.EPOCH.date().year, c.EPOCH.date().month, c.EPOCH.date().day)
+        if 'price' in params:
+            if pc.expiration_date == '':
+                pass
+            else:
+                year, month, day = pc.expiration_date.split("-")
+                pc.expiration_date = datetime(int(year), int(month), int(day)).replace(tzinfo=UTC)
+            if pc.price == '':
```

Comment:
One thing that would help code readability is if you created two sets of `if / elif` blocks: one for "data management" (e.g., auto-generating the promo code), and one for "validation" (e.g., making sure they enter a price). You could even put the 'data management' block _after_ the `if message == ''` block below (which itself should be written as `if not message`)